File: 1013-fibonacci-number/fibonacci-number.cpp
class Solution {
public:
    int dp[31];
    int find(int n)
    {
        if(n<2)
        {
            return n;
        }
        if(dp[n]!=-1)
        {
            return dp[n];
        }
        return dp[n]=find(n-1)+find(n-2);
    }
    int fib(int n) {
        memset(dp,-1,sizeof(dp));
        return find(n);
    }
};

File: 1013-fibonacci-number/fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n==0 || n==1)
        {
            return n;
        }
        int f0=0,f1=1,result=0;
        for(int i=2;i<=n;i++)
        {
            result=f0+f1;
            f0=f1;
            f1=result;
        }
        return result;
    }
}